driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedCrudHelper.java

```diff
@@ -131,14 +142,78 @@ OperationResult executeListDatabases(final BsonDocument operation) {
                 new BsonArray(client.listDatabases(BsonDocument.class).into(new ArrayList<>())));
     }
 
+    OperationResult executeListCollections(final BsonDocument operation) {
+        MongoDatabase database = entities.getDatabase(operation.getString("object").getValue());
+
+        BsonDocument arguments = operation.getDocument("arguments");
+        ListCollectionsIterable<BsonDocument> iterable = database.listCollections(BsonDocument.class);
+        for (Map.Entry<String, BsonValue> cur : arguments.entrySet()) {
+            switch (cur.getKey()) {
+                case "filter":
+                    iterable.filter(cur.getValue().asDocument());
+                    break;
+                case "batchSize":
+                    iterable.batchSize(cur.getValue().asNumber().intValue());
+                    break;
+                default:
+                    throw new UnsupportedOperationException("Unsupported argument: " + cur.getKey());
+            }
+        }
+
+        return resultOf(() ->
+                new BsonArray(iterable.into(new ArrayList<>())));
+    }
+
+    OperationResult executeListIndexes(final BsonDocument operation) {
+        MongoCollection<BsonDocument> collection = entities.getCollection(operation.getString("object").getValue());
+        BsonDocument arguments = operation.getDocument("arguments");
+        ListIndexesIterable<BsonDocument> iterable = collection.listIndexes(BsonDocument.class);
+        for (Map.Entry<String, BsonValue> cur : arguments.entrySet()) {
+            //noinspection SwitchStatementWithTooFewBranches
```

Comment:
This appears to be a useless IDEA check that we could globally disable in the settings.

Reply:
OK, I can look to see how to do that.  Until then, are you asking that I remove this?

Reply:
If we agree that the check is not helpful in general then yes, its better to remove the comment and anyone may just disable the check for himself via `Alt + Enter` | Inspection ... options | Disable inspection. If we do not agree on this, then the comment should stay.

Reply:
There are six other occurrences of this already in the unified runner.  So let's leave this until we decide to remove them all.